core/CronArchive.php

```diff
@@ -894,6 +894,24 @@ private function invalidateWithSegments($idSites, $date, $period, $_forceInvalid
                 if ($this->canWeSkipInvalidatingBecauseThereIsAUsablePeriod($params, $doNotIncludeTtlInExistingArchiveCheck)) {
                     $this->logger->debug('  Found usable archive for {archive}, skipping invalidation.', ['archive' => $params]);
                 } else {
+                    if (empty($this->segmentArchiving)) {
+                        // might not be initialised if init is not called
+                        $this->segmentArchiving = StaticContainer::get(SegmentArchiving::class);
+                    }
```

Comment:
Do you know where init() is not being called? I think we should always be calling it if the use of the object requires it, rather than conditionally getting it like this.

Reply:
I saw this in an other method in this file, that's why I did the same. I did not check if that's true or not.

Reply:
I see. Well it's a minor issue, so it's not a big deal if you don't take care of it.